lib/dao.js

```diff
@@ -2689,6 +2689,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   Model.notifyObserversOf('before save', context, function(err, ctx) {
     if (err) return cb(err);
 
+    if (ctx.instance[pkName] !== id && !(Model._warned || {}).cannotOverwritePKInBeforeSaveHook) {
+      Model._warned = { cannotOverwritePKInBeforeSaveHook: true };
+      console.warn('WARNING: id property cannot be changed from ' +
+        id + ' to ' + inst[pkName] + ' in \'before save\' operation hook');
```

Comment:
Please include model name (`Model.modelName`) in the message. Otherwise it's difficult to figure out where is the source code triggering this warning without running the debugger.